oval-schemas/linux-definitions-schema.xsd

```diff
@@ -2718,6 +2718,110 @@
+             <sch:rule context="linux:auditdline_test/linux:state">
+               <sch:assert test="@state_ref=ancestor::oval-def:oval_definitions/oval-def:states/linux:auditdline_state/@id"><sch:value-of select="../@id"/> - the state child element of a auditdline_test must reference a auditdline_state</sch:assert>
+             </sch:rule>
+           </sch:pattern>
+         </xsd:appinfo>
+       </xsd:annotation>
+       <xsd:complexType>
+         <xsd:complexContent>
+           <xsd:extension base="oval-def:TestType">
+             <xsd:sequence>
+               <xsd:element name="object" type="oval-def:ObjectRefType" />
+               <xsd:element name="state" type="oval-def:StateRefType" minOccurs="0" maxOccurs="unbounded"/>
+             </xsd:sequence>
+           </xsd:extension>
+         </xsd:complexContent>
+       </xsd:complexType>
+     </xsd:element>
+     <xsd:element name="auditdline_object" substitutionGroup="oval-def:object">
+       <xsd:annotation>
+         <xsd:documentation>The auditdline_object element is used by a auditdline_test to define the object to be evaluated. Each object extends the standard ObjectType as defined in the oval-definitions-schema and one should refer to the ObjectType description for more information. The common set element allows complex objects to be created using filters and set logic. Again, please refer to the description of the set element in the oval-definitions-schema.</xsd:documentation>
+         <xsd:documentation>A auditdline_object consists of an filter_key entity that is the same as the -k parameter of the auditctl -l command.</xsd:documentation>
+         <xsd:appinfo>
+           <sch:pattern id="linux_auditdline_object_verify_filter_state">
+             <sch:rule context="linux:auditdline_object//oval-def:filter">
+               <sch:let name="parent_object" value="ancestor::linux:auditdline_object"/>
+               <sch:let name="parent_object_id" value="$parent_object/@id"/>
+               <sch:let name="state_ref" value="."/>
+               <sch:let name="reffed_state" value="ancestor::oval-def:oval_definitions/oval-def:states/*[@id=$state_ref]"/>
+               <sch:let name="state_name" value="local-name($reffed_state)"/>
+               <sch:let name="state_namespace" value="namespace-uri($reffed_state)"/>
+               <sch:assert test="(($state_namespace='http://oval.mitre.org/XMLSchema/oval-definitions-5#linux') and ($state_name='auditdline_state'))">State referenced in filter for <sch:value-of select="name($parent_object)"/> '<sch:value-of select="$parent_object_id"/>' is of the wrong type. </sch:assert>
+             </sch:rule>
+           </sch:pattern>
+         </xsd:appinfo>
+       </xsd:annotation>
+       <xsd:complexType>
+         <xsd:complexContent>
+           <xsd:extension base="oval-def:ObjectType">
+             <xsd:sequence>
+               <xsd:choice>
+                 <xsd:element ref="oval-def:set"/>
+                 <xsd:sequence>
```

Comment:
Unfortunately, the `key` is largely not useful to see if things are correct. I've found that different users will throw in different keys for their own internal processing consistency across platforms/releases and so I think that it should be ignored.

`field`, `syscall`, and `permission` definitely make sense.

Reply:
Thank you for this feedback.  
I think this is a good idea to improve the `auditdline_test`.  
This will allow the test to be simpler to implement and to use !